docs/developer-guide/contributing.md

```diff
@@ -156,6 +156,29 @@ make: *** [Makefile:386: verify-kube-connect] Error 1
 
 you should edit your `~/.kube/config` and modify the `server` option to point to your correct K8s API (as described above).
 
+### Using k3d
+
+k3d is a lightweight wrapper to run [k3s](https://github.com/rancher/k3s), a minimal Kubernetes distribution, in docker. Because it's running in a docker container, you're dealing with docker's internal networking rules when using k3d. A typical Kubernetes cluster running on your local machine is part of the same network that you're on so you can access it using **kubectl**. However, a Kubernetes cluster running within a docker container (in this case, the one launched by make) cannot access 0.0.0.0 from inside the container itself, when 0.0.0.0 is a network resource outside the container itself (and/or the container's network). This is the cost of a fully self-contained, disposable Kubernetes cluster. The following steps should help with a successful `make verify-kube-connect` execution.
```

Comment:
In the first point of  https://github.com/argoproj/argo-cd/blob/master/docs/developer-guide/contributing.md#before-you-start it has been mentioned that a minimal Kubernetes cluster k3s is enough for dev environment.  From - https://rancher.com/docs/k3s/latest/en/installation/installation-requirements/#operating-systems i can see k3s is only supported on linux distribution, so on mac and windows we need a third party tool like k3d, VirtualBox etc which can provide a containerize/virtual linux env to install k3s.

Instead of specifying a tool to install k3s, IMO just create a hyper link k3d in the first point of https://github.com/argoproj/argo-cd/blob/master/docs/developer-guide/contributing.md#before-you-start as one of containerize env provider and ofcourse for connectivity you need to provide ip address configuration in k3d containerize env.